src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcAnalyzer.scala

```diff
@@ -1,287 +0,0 @@
- *
- */
-
-package com.amazon.deequ.analyzers.jdbc
-
-import java.sql.{JDBCType, ResultSet}
-
-import com.amazon.deequ.analyzers.State
-import com.amazon.deequ.analyzers.runners._
-import com.amazon.deequ.metrics.{DoubleMetric, Entity, Metric}
-
-import java.sql.Types._
-
-import scala.util.{Failure, Success}
-
-trait JdbcAnalyzer[S <: State[_], +M <: Metric[_]] {
-
-  /**
-    * Compute the state (sufficient statistics) from the data
-    * @param table database table
-    * @return
-    */
-  def computeStateFrom(table: Table): Option[S]
-
-  /**
-    * Compute the metric from the state (sufficient statistics)
-    * @param state wrapper holding a state of type S (required due to typing issues...)
-    * @return
-    */
-  def computeMetricFrom(state: Option[S]): M
-
-  /**
-    * A set of assertions that must hold on the database table
-    * @return
-    */
-  def preconditions: Seq[Table => Unit] = {
-    Seq.empty
-  }
-
-  def validatePreconditions(table: Table): Unit = {
-    val exceptionOption = Preconditions.findFirstFailing(table, this.preconditions)
-    if (exceptionOption.nonEmpty) {
-      throw exceptionOption.get
-    }
-  }
-
-  def calculate(table: Table): M = {
-
-    try {
-      validatePreconditions(table)
-      val state = computeStateFrom(table)
-      computeMetricFrom(state)
-    } catch {
-      case error: Exception => toFailureMetric(error)
-    }
-  }
-
-  private[deequ] def toFailureMetric(failure: Exception): M
-}
-
-/** Base class for analyzers that require to group the data by specific columns */
-abstract class GroupingAnalyzer[S <: State[_], +M <: Metric[_]] extends JdbcAnalyzer[S, M] {
-
-  /** The columns to group the data by */
-  def groupingColumns(): Seq[String]
-
-  /** Ensure that the grouping columns exist in the data */
-  override def preconditions: Seq[Table => Unit] = {
-    groupingColumns().map { name => Preconditions.hasColumn(name) } ++ super.preconditions
-  }
-}
-
-/** Helper method to check conditions on the schema of the data */
-object Preconditions {
-
-  private[this] val numericDataTypes =
-    Set(BIGINT, DECIMAL, DOUBLE, FLOAT, INTEGER, NUMERIC, SMALLINT, TINYINT)
-
-  /* Return the first (potential) exception thrown by a precondition */
-  def findFirstFailing(
-                        table: Table,
-                        conditions: Seq[Table => Unit])
-  : Option[Exception] = {
-
-    conditions.map { condition =>
-      try {
-        condition(table)
-        None
-      } catch {
-        /* We only catch exceptions, not errors! */
-        case e: Exception => Some(e)
-      }
-    }
-      .find { _.isDefined }
-      .flatten
-  }
-
-  /** Specified table exists in the data */
-  def hasTable(): Table => Unit = { table =>
-
-    val connection = table.jdbcConnection
-
-    val metaData = connection.getMetaData
-    val result = metaData.getTables(null, null, null, Array[String]("TABLE"))
-
-    var hasTable = false
-
-    while (result.next()) {
```

Comment:
I meant the exists function of an iterator https://docs.scala-lang.org/overviews/collections/iterators.html It will also iterate over the results, but the code is much nicer to read than manually programming that